R/colourInput.R

```diff
@@ -221,10 +221,16 @@ formatHEXsingle <- function(x) {
   if (x %in% grDevices::colors()) {
     x <- do.call(grDevices::rgb, as.list(grDevices::col2rgb(x) / 255))
   }
-  if (!grepl("^#?([[:xdigit:]]{3}|[[:xdigit:]]{6})$", x)) {
+  if (!grepl("^#?([[:xdigit:]]{3}|[[:xdigit:]]{6}|[[:xdigit:]]{8})$", x)) {
     stop(sprintf("%s is not a valid colour", x), call. = FALSE)
   }
 
+  # check whether it is 6-digit hex code with alpha.
```

Comment:
Related to my comment below, this code doesn't work:

```
plot(rnorm(100), col = "FFFFFF")
Error in plot.xy(xy, type, ...) : invalid color name 'FFFFFF'
```
Same with alpha included. But maybe I'm missing the point and this is an R thing and you are talking about JS, and in JS it is allowed?